testsuite/tests/ocaml_api/general/main.ml

```diff
@@ -217,6 +217,22 @@ let test_node () =
     (FooNode.for_all aux root) ;
   Format.printf "@[<v>============================@ @ @]"
 
+let test_lookup_with_kind () = 
+  Format.printf "@[<v>========== LOOKUP ==========@ @]" ;
+  let ctx = AnalysisContext.create () in
+  let u = AnalysisContext.get_from_file ~reparse:true ctx "foo.txt" in
+  print_exit_if_diags u ;
+  let root = root_exn u in
+  let lookup_1 = FooNode.lookup_with_kind Sequence root {Sloc.line=1; Sloc.column=25} in
```

Comment:
SInce the goal of lookup_with_kind is to have a correct type for the returned node, I suggest to add a typing information for lookup_1. Something like:
let lookup_1 : Sequence.t option.

Reply:
Done in b0daa20